src/roles/dullahan.py

```diff
@@ -5,76 +5,73 @@
 
 import src.settings as var
 from src.utilities import *
-from src import debuglog, errlog, plog
-from src.decorators import cmd, event_listener
+from src.functions import get_players, get_target
+from src import users, debuglog, errlog, plog
+from src.decorators import command, event_listener
 from src.messages import messages
 from src.events import Event
 import botconfig
 
-KILLS = {} # type: Dict[str, str]
-TARGETS = {} # type: Dict[str, Set[str]]
+KILLS = {} # type: Dict[users.User, users.User]
+TARGETS = {} # type: Dict[users.User, Set[users.User]]
 
-@cmd("kill", chan=False, pm=True, playing=True, silenced=True, phases=("night",), roles=("dullahan",))
-def dullahan_kill(cli, nick, chan, rest):
+@command("kill", chan=False, pm=True, playing=True, silenced=True, phases=("night",), roles=("dullahan",))
+def dullahan_kill(var, wrapper, message):
     """Kill someone at night as a dullahan until everyone on your list is dead."""
-    if not TARGETS[nick] & set(list_players()):
-        pm(cli, nick, messages["dullahan_targets_dead"])
+    if not TARGETS[wrapper.source] & set(get_players()):
+        wrapper.pm(messages["dullahan_targets_dead"])
         return
 
-    victim = get_victim(cli, nick, re.split(" +",rest)[0], False)
-    if not victim:
+    target = get_target(var, wrapper, re.split(" +", message)[0])
+    if not target:
         return
 
-    if victim == nick:
-        pm(cli, nick, messages["no_suicide"])
+    if target is wrapper.source:
+        wrapper.pm(messages["no_suicide"])
         return
 
-    orig = victim
-    evt = Event("targeted_command", {"target": victim, "misdirection": True, "exchange": True})
-    evt.dispatch(cli, var, "kill", nick, victim, frozenset({"detrimental"}))
+    orig = target
+    evt = Event("targeted_command", {"target": target.nick, "misdirection": True, "exchange": True})
+    evt.dispatch(wrapper.client, var, "kill", wrapper.source.nick, target.nick, frozenset({"detrimental"}))
     if evt.prevent_default:
         return
-    victim = evt.data["target"]
+    target = users._get(evt.data["target"]) # FIXME: Need to fix once targeted_command uses the new API
 
-    KILLS[nick] = victim
+    KILLS[wrapper.source] = target
 
-    msg = messages["wolf_target"].format(orig)
-    pm(cli, nick, messages["player"].format(msg))
+    wrapper.pm(messages["player"].format(messages["wolf_target"].format(orig)))
 
-    debuglog("{0} ({1}) KILL: {2} ({3})".format(nick, get_role(nick), victim, get_role(victim)))
+    debuglog("{0} ({1}) KILL: {2} ({3})".format(wrapper.source, get_role(wrapper.source.nick), target, get_role(target.nick)))
```

Comment:
change get_role(wrapper.source.nick) to a hardcoded string "dullahan"

If dullahan is applied as a template this would cause very wacky debug logs. This will be something that gets adjusted going forwards; debug logs should always hardcode the roles that are triggering the logs, as such roles are already known.